From d556862e731b9f31993d1ec67d6edb9f54e460bd Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 11:59:04 -0400
Subject: [infiniband] core: Add support for RDMA_NODE_USNIC_UDP

Message-id: <3fea350d7af51070053af47e839901dc215f6661.1410425016.git.avadai@redhat.com>
Patchwork-id: 93583
O-Subject: [Patch RHEL7.1 V1 008/197] IB/core: Add support for RDMA_NODE_USNIC_UDP
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Upinder Malhi <umalhi@cisco.com>

Add the complementary RDMA_NODE_USNIC_UDP for RDMA_TRANSPORT_USNIC_UDP.

Signed-off-by: Upinder Malhi <umalhi@cisco.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 5db5765e255de4072eb0e35facfeafce53af001b)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/core/sysfs.c b/drivers/infiniband/core/sysfs.c
index faad2ca..7d3292c 100644
--- a/drivers/infiniband/core/sysfs.c
+++ b/drivers/infiniband/core/sysfs.c
@@ -613,6 +613,7 @@ static ssize_t show_node_type(struct device *device,
  case RDMA_NODE_IB_CA:   return sprintf(buf, "%d: CA\n", dev->node_type);
  case RDMA_NODE_RNIC:   return sprintf(buf, "%d: RNIC\n", dev->node_type);
  case RDMA_NODE_USNIC:   return sprintf(buf, "%d: usNIC\n", dev->node_type);
+ case RDMA_NODE_USNIC_UDP: return sprintf(buf, "%d: usNIC UDP\n", dev->node_type);
  case RDMA_NODE_IB_SWITCH: return sprintf(buf, "%d: switch\n", dev->node_type);
  case RDMA_NODE_IB_ROUTER: return sprintf(buf, "%d: router\n", dev->node_type);
  default:    return sprintf(buf, "%d: <unknown>\n", dev->node_type);
diff --git a/drivers/infiniband/core/verbs.c b/drivers/infiniband/core/verbs.c
index c1cfeda..3ac7951 100644
--- a/drivers/infiniband/core/verbs.c
+++ b/drivers/infiniband/core/verbs.c
@@ -118,6 +118,8 @@ rdma_node_get_transport(enum rdma_node_type node_type)
  case RDMA_NODE_RNIC:
   return RDMA_TRANSPORT_IWARP;
  case RDMA_NODE_USNIC:
+  return RDMA_TRANSPORT_USNIC;
+ case RDMA_NODE_USNIC_UDP:
   return RDMA_TRANSPORT_USNIC_UDP;
  default:
   BUG();
diff --git a/include/rdma/ib_verbs.h b/include/rdma/ib_verbs.h
index e646ff8..6793f32 100644
--- a/include/rdma/ib_verbs.h
+++ b/include/rdma/ib_verbs.h
@@ -70,6 +70,7 @@ enum rdma_node_type {
  RDMA_NODE_IB_ROUTER,
  RDMA_NODE_RNIC,
  RDMA_NODE_USNIC,
+ RDMA_NODE_USNIC_UDP,
 };
 
 enum rdma_transport_type {
-- 
1.7.1